### DISCUSSION TITLE
Home assistant

### DISCUSSION THREAD

**Discussion by visualapproach** [visualapproach/WiFi-remote-for-Bestway-Lay-Z-SPA#540] [Q&A]
Hi all! So, HA has an opinion on the 'time to ready' sensor. It's usually a numeric value that tells the number of hours until set temperature is reached. But there is also two states 'Never' and 'Already'. In the code it's represented as -1 and -2 but translated to a string before sending to MQTT. What would be the best way to handle this? If I send everything as a string we wouldn't be able to see graphs easily. Replacing the 'never/already' with numbers will work but is not that obvious what it means.

**Comment by acolod** [comment]
Hi @visualapproach.  First of all, THANK YOU for all of your work on this. I recently purchased the Helsinki spa which has wifi/app control listed in the Amazon listing - obviously that was a lie!  I found this project and have built the controller and have it all up and running. What magic!  

I then set up HA and have been working on automations, and this issue has been driving me nuts!!!  I am not a very experienced coder, but have tired a few ways to deal with it in HA (not touching the source code, I'm terrified of breaking my controller lol). I've tried some python scripts and some complex automations with no real luck.  My current attempt to deal with this is using a template although it's not quite working yet:
```
sensor:
  - platform: template
    sensors:
      time_to_ready_numeric:
        friendly_name: "Time to Ready (Numeric)"
        value_template: >
          {% if is_state('sensor.time_to_ready', 'never') %}
            -1
          {% elif is_state('sensor.time_to_ready', 'already') %}
            0
          {% else %}
            {{ states('sensor.time_to_ready') }}
          {% endif %}
```

I realize from your post that you don't want to just convert them to numbers, however I would think you could create another template sensor to display the text versions of those values:

```
sensor:
  - platform: template
    sensors:
      time_to_ready_display:
        friendly_name: "Time to Ready (Display)"
        value_template: >
          {% if is_state('sensor.time_to_ready_numeric', '-1') %}
            'never'
          {% elif is_state('sensor.time_to_ready_numeric', '0') %}
            'already'
          {% else %}
            {{ states('sensor.time_to_ready_numeric') }}
          {% endif %}
```

That said, I don't really know what the heck I'm doing and learning as I go so maybe this is redundant and can be handled better.  Happy to help work through this as most of my scheduling automations rely on time to ready so they're all currently broken!

**Reply by visualapproach** [reply]
Thanks for your response! I haven't given it a deeper thought yet. I just saw HA complaining about it and wanted some feedback if there was any thoughts about it. Will address it as soon as I can

**Reply by acolod** [reply]
Got it! Would certainly support a code side solution that plays well with HA.  Would it work to use separate MQTT topics for the numeric and textual representations of the "time to ready" value. For example, one topic could be used to send the numeric value (including -1 and -2 for "never" and "already"), and another topic could be used to send the textual representation ("never", "already", or the number of hours as a string). Would still allow graphing?

**Reply by visualapproach** [reply]
Thanks for your ideas, BETA is updated with 
T2R - only numeric, including -1 and -2
RS (Ready State) - "Never"/"Ready"/"Not Ready" strings

**Reply by acolod** [reply]
This is awesome, much thanks! Working well with ha now it seems
